### PR TITLE
drivers: serial: pl011: Remove busy wait in Ambiq UART initiate

### PR DESCRIPTION
**Pull request by BryanZhuAM**
Ambiq UART requires specific busy wait during initialization for propagating powering control registers, original k_busy_wait() used here generated a dead loop because k_busy_wait() relays on timer, who's driver is initialized after UART(UART init in PRE_KERNEL_1, timer init in PRE_KERNEL_2), replace k_busy_wait() with checking power status register is more suitable here.

This issue impacts the OS's normal startup in apollo4p SOC， even samples:hello_world can not work and stuck in k_busy_wait()

### CODE REVIEW

**Review by fkokosinski — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by kgugala** [comment]
@zchen-ambiq the PR has already two acks, so you just need to wait for someone with merge rights to merge it (or for some new change requests if there will be any)

**Pull request merged**
